### ISSUE TITLE
Doesn't detect Promise.allSettled is unavailable in Safari 12

### ISSUE THREAD

**Issue by Bogidon** [amilajack/eslint-plugin-compat#297]
- version: 3.5.1
- .browserlistrc:
```
Chrome >= 72
Firefox >= 72
Safari >= 12
Edge >= 79
```

Using `Promise.allSettled` does not produce an error for Safari. [caniuse link](https://caniuse.com/#feat=mdn-javascript_builtins_promise_allsettled)

Not sure if this is the right place to report.

**Comment by brettz9** [comment]
I think this should be closed in favor of the more generic #206 (letting that apply to all ES features).

**Comment by amilajack** [comment]
@brettz9 After some recent changes, We can now easily lint ES APIs as well. We still dont lint for syntactic ES features because we assume those will be transpiled by babel so the user doesnt have to worry about their compatibility or not.

**Comment by brettz9** [comment]
Cool, I see. And there is `es-compat` anyways for syntax checking for those who want to check it (e.g., to see if it is even necessary to transpile).
